src/components/utils/src/threads/thread_posix.cc

```diff
@@ -275,10 +278,13 @@ void Thread::join() {
 }
 
 Thread::~Thread() {
-  finalized_ = true;
-  stopped_ = true;
-  join();
-  pthread_join(handle_, NULL);
+  if (!finalized_) {
+    ThreadCounter::Decrement();
+    finalized_ = true;
+    stopped_ = true;
```

Comment:
@orhan-mehmedov, yes I saw this, i just added if case in destructor, didn't remove this, probably there is some case, were it used. Didn't checked.